Library/Homebrew/cmd/upgrade.rb

```diff
@@ -104,6 +105,35 @@ def upgrade
         onoe "#{f}: #{e}"
       end
     end
+
+    unless Homebrew.failed?
```

Comment:
`return if Homebrew.failed?`

Reply:
We cannot return yet, as we have not printed the caveats of the upgraded casks yet.

Reply:
@amyspark sorry, misread, you're right!